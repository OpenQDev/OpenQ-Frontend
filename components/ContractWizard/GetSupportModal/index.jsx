import Image from 'next/image';
import React, { useEffect, useRef } from 'react';
import ModalDefault from '../../Utils/ModalDefault';

const GetSupportModal = ({ wizardVisibility, modalVisibility }) => {
  // Refs
  const modal = useRef();

  useEffect(() => {
    // Courtesy of https://stackoverflow.com/questions/32553158/detect-click-outside-react-component
    function handleClickOutside(event) {
      if (modal.current && !modal.current.contains(event.target)) {
        modalVisibility(false);
        wizardVisibility(false);
      }
    }
    // Bind the event listener
    {
      document.addEventListener('mousedown', handleClickOutside);
    }
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [modal]);

  const resetState = () => {
    modalVisibility(false);
  };

  const btn = (
    <a
      href={'https://discord.gg/puQVqEvVXn'}
      className='flex items-center gap-2 btn-primary'
      target={'_blank'}
      rel='noopener noreferrer'
    >
      <div data-testid='link' className='flex items-center gap-2'>
        <Image src={'/social-icons/discord.svg'} width={24} height={24} alt='discord-logo' />
        Get Help!
      </div>
    </a>
  );

  return (
    <>
      <ModalDefault
        title={"We didn't find a suitable contract"}
        footerRight={btn}
        setShowModal={wizardVisibility}
        resetState={resetState}
      >
        <h3 className='text-lg pt-4'>
          Don't worry, you can help us and describe more precisely what kind of contract you want. If we don't have this
          one yet, we'll be happy to get to it next.
        </h3>
        <p className='pt-4 pr-4'>
          Join our Discord and write us in the support channel or send us an email at riccardo@openq.dev
        </p>
      </ModalDefault>
    </>
  );
};

export default GetSupportModal;
