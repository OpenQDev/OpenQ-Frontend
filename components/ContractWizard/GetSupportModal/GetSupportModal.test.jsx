/**
 * @vi-environment jsdom
 */
import React from 'react';
import GetSupportModal from '.';
import renderer from 'react-test-renderer';

describe('GetSupportModal', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<GetSupportModal modalVisibility={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
