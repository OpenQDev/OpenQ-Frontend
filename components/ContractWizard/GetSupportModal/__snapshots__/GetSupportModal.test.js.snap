// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`GetSupportModal should match DOM Snapshot 1`] = `
<div>
  <div
    className="flex justify-center items-end sm:items-center overflow-x-hidden overflow-y-auto fixed inset-0 z-[53]"
  >
    <div
      className="flex w-full sm:w-[480px] h-[320px] border border-gray-700 sm:rounded-sm bg-[#161B22]"
    >
      <div
        className="flex flex-col relative w-full"
      >
        <button
          className="absolute top-4 right-4 cursor-pointer"
          data-testid="cross"
          onClick={[Function]}
        >
          <svg
            className="fill-muted hover:fill-blue-400 w-4 h-4"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
              fillRule="evenodd"
            />
          </svg>
        </button>
        <div
          className="py-2 px-4 border-b border-gray-700 text-2xl w-full h-[50px]"
        >
          We didn't find a suitable contract
        </div>
        <div
          className="py-2 px-4 w-full h-[220px] overflow-x-hidden overflow-y-auto"
        >
          <h3
            className="text-lg pt-4"
          >
            Don't worry, you can help us and describe more precisely what kind of contract you want. If we don't have this one yet, we'll be happy to get to it next.
          </h3>
          <p
            className="pt-4 pr-4"
          >
            Join our Discord and write us in the support channel or send us an email at riccardo@openq.dev
          </p>
        </div>
        <div
          className="flex justify-end  py-2 px-4 border-t border-gray-700 w-full h-[50px]"
        >
          <div />
          <div>
            <a
              className="flex items-center gap-2 btn-primary"
              href="https://discord.gg/puQVqEvVXn"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                className="flex items-center gap-2"
                data-testid="link"
              >
                <img
                  alt="discord-logo"
                  data-nimg="1"
                  decoding="async"
                  height={24}
                  loading="lazy"
                  onError={[Function]}
                  onLoad={[Function]}
                  src="/social-icons/discord.svg"
                  style={
                    {
                      "color": "transparent",
                    }
                  }
                  width={24}
                />
                Get Help!
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="bg-overlay z-10 fixed inset-0"
  />
</div>
`;
